From 3cd12ed38eee45fb6dd5196e5c72533765bae492 Mon Sep 17 00:00:00 2001
From: Jake <jake@fulu.org>
Date: Thu, 9 Jul 2026 16:40:36 +0100
Subject: [PATCH] Hide Add as friend/foe links when friending is disabled

---
 UserProfile/includes/UserProfilePage.php | 55 +++++++++++++-----------
 1 file changed, 29 insertions(+), 26 deletions(-)

diff --git a/UserProfile/includes/UserProfilePage.php b/UserProfile/includes/UserProfilePage.php
index 575474e1..6f3063ee 100644
--- a/UserProfile/includes/UserProfilePage.php
+++ b/UserProfile/includes/UserProfilePage.php
@@ -1110,34 +1110,37 @@ class UserProfilePage extends Article {
 			$profileLinks['user-watchlist'] =
 				'<a href="' . htmlspecialchars( $watchlist->getFullURL() ) . '">' . wfMessage( 'user-watchlist' )->escaped() . '</a>';
 		} elseif ( $this->viewingUser->isRegistered() ) {
-			// Support for friendly-by-default URLs (T191157)
-			$add_friend = SpecialPage::getTitleFor(
-				'AddRelationship',
-				$this->profileOwner->getName() . '/friend'
-			);
-			$add_foe = SpecialPage::getTitleFor(
-				'AddRelationship',
-				$this->profileOwner->getName() . '/foe'
-			);
-			$remove_relationship = SpecialPage::getTitleFor(
-				'RemoveRelationship',
-				$this->profileOwner->getName()
-			);
+			global $wgFriendingEnabled;
+			if ( $wgFriendingEnabled ) {
+				// Support for friendly-by-default URLs (T191157)
+				$add_friend = SpecialPage::getTitleFor(
+					'AddRelationship',
+					$this->profileOwner->getName() . '/friend'
+				);
+				$add_foe = SpecialPage::getTitleFor(
+					'AddRelationship',
+					$this->profileOwner->getName() . '/foe'
+				);
+				$remove_relationship = SpecialPage::getTitleFor(
+					'RemoveRelationship',
+					$this->profileOwner->getName()
+				);
 
-			if ( $relationship == false ) {
-				$profileLinks['user-add-friend'] =
-					'<a href="' . htmlspecialchars( $add_friend->getFullURL() ) . '" rel="nofollow">' . wfMessage( 'user-add-friend' )->escaped() . '</a>';
+				if ( $relationship == false ) {
+					$profileLinks['user-add-friend'] =
+						'<a href="' . htmlspecialchars( $add_friend->getFullURL() ) . '" rel="nofollow">' . wfMessage( 'user-add-friend' )->escaped() . '</a>';
 
-				$profileLinks['user-add-foe'] =
-					'<a href="' . htmlspecialchars( $add_foe->getFullURL() ) . '" rel="nofollow">' . wfMessage( 'user-add-foe' )->escaped() . '</a>';
-			} else {
-				if ( $relationship == 1 ) {
-					$profileLinks['user-remove-friend'] =
-						'<a href="' . htmlspecialchars( $remove_relationship->getFullURL() ) . '">' . wfMessage( 'user-remove-friend' )->escaped() . '</a>';
-				}
-				if ( $relationship == 2 ) {
-					$profileLinks['user-remove-foe'] =
-						'<a href="' . htmlspecialchars( $remove_relationship->getFullURL() ) . '">' . wfMessage( 'user-remove-foe' )->escaped() . '</a>';
+					$profileLinks['user-add-foe'] =
+						'<a href="' . htmlspecialchars( $add_foe->getFullURL() ) . '" rel="nofollow">' . wfMessage( 'user-add-foe' )->escaped() . '</a>';
+				} else {
+					if ( $relationship == 1 ) {
+						$profileLinks['user-remove-friend'] =
+							'<a href="' . htmlspecialchars( $remove_relationship->getFullURL() ) . '">' . wfMessage( 'user-remove-friend' )->escaped() . '</a>';
+					}
+					if ( $relationship == 2 ) {
+						$profileLinks['user-remove-foe'] =
+							'<a href="' . htmlspecialchars( $remove_relationship->getFullURL() ) . '">' . wfMessage( 'user-remove-foe' )->escaped() . '</a>';
+					}
 				}
 			}
 
