From 4e0b4f3551c15963ffa130cced1025e4b1a36893 Mon Sep 17 00:00:00 2001
From: Jake <jake@fulu.org>
Date: Thu, 9 Jul 2026 19:23:09 +0100
Subject: [PATCH] Only show "Send a gift" link to giftadmin holders

---
 UserProfile/includes/UserProfilePage.php | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/UserProfile/includes/UserProfilePage.php b/UserProfile/includes/UserProfilePage.php
index 575474e1..209810a8 100644
--- a/UserProfile/includes/UserProfilePage.php
+++ b/UserProfile/includes/UserProfilePage.php
@@ -1150,9 +1150,11 @@ class UserProfilePage extends Article {
 					] ) ) . '" rel="nofollow">' .
 					wfMessage( 'user-send-message' )->escaped() . '</a>';
 			}
-			$profileLinks['user-send-gift'] =
-				'<a href="' . htmlspecialchars( $give_gift->getFullURL() ) . '" rel="nofollow">' .
-				wfMessage( 'user-send-gift' )->escaped() . '</a>';
+			if ( $this->viewingUser->isAllowed( 'giftadmin' ) ) {
+				$profileLinks['user-send-gift'] =
+					'<a href="' . htmlspecialchars( $give_gift->getFullURL() ) . '" rel="nofollow">' .
+					wfMessage( 'user-send-gift' )->escaped() . '</a>';
+			}
 		}
 
 		$profileLinks['user-contributions'] =
